src/Spryker/Zed/AppKernel/Business/Writer/ConfigWriter.php

```diff
@@ -70,21 +73,65 @@ public function saveConfig(AppConfigTransfer $appConfigTransfer): AppConfigRespo
      *
      * @return \Generated\Shared\Transfer\AppConfigTransfer
      */
-    protected function doSaveAppConfig(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
+    protected function mergeWithExistingConfiguration(AppConfigTransfer $appConfigTransfer): AppConfigTransfer
     {
-        if ($this->configurationBeforeSavePlugin) {
-            $appConfigTransfer = $this->configurationBeforeSavePlugin->beforeSave($appConfigTransfer);
+        try {
+            /** @var \Generated\Shared\Transfer\AppConfigTransfer $existingAppConfigTransfer */
+            $existingAppConfigTransfer = $this->appKernelRepository->findAppConfigByCriteria(
+                (new AppConfigCriteriaTransfer())->setTenantIdentifier($appConfigTransfer->getTenantIdentifierOrFail()),
+                new AppConfigTransfer(),
```

Comment:
`new AppConfigTransfer()`
why it's like this?  Seems it's a leftover, currently every app uses this transfer and it should not be provided as input.

Reply:
Can we re-evaluate this later?

Reply:
but am I right?